projects/core/src/features-config/feature-toggles/feature-toggles-providers.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { FactoryProvider, ValueProvider } from '@angular/core';
+import {
+  DefaultFeatureTogglesChunk,
+  FeatureToggles,
+  FeatureTogglesChunk,
+} from './feature-toggles-tokens';
+
+export type FeatureTogglesFactory = (...props: any[]) => FeatureToggles;
+
+/**
+ * Helper function to provide FeatureTogglesChunk token
+ *
+ * To provide DefaultFeatureTogglesChunk flags in libraries, provideFeatureTogglesChunk should be used instead.
+ *
+ * @param flags FeatureToggles object to merge with the global flags
+ */
+export function provideFeatureToggles(
+  flags: FeatureToggles = {},
+  defaultFlags = false
+): ValueProvider {
+  return {
+    provide: defaultFlags ? DefaultFeatureTogglesChunk : FeatureTogglesChunk,
+    useValue: flags,
+    multi: true,
+  };
+}
+
+/**
+ * Helper function to provide FeatureTogglesChunk with a factory function
+ *
+ * To provide DefaultFeatureTogglesChunk flags in libraries provideDefaultFeatureTogglesFactory should be used instead.
+ *
+ * @param featureTogglesFactory Factory Function that will generate flags object
+ * @param deps Optional dependencies to a factory function
+ */
+export function provideFeatureTogglesFactory(
+  featureTogglesFactory: FeatureTogglesFactory,
+  defaultFlags = false
+): FactoryProvider {
+  return {
+    provide: defaultFlags ? DefaultFeatureTogglesChunk : FeatureTogglesChunk,
+    useFactory: featureTogglesFactory,
+    multi: true,
+  };
+}
+
+/**
+ * Helper function to provide DefaultFeatureTogglesChunk token
+ *
+ * @param flags FeatureToggles object to merge with the default flags
+ */
+export function provideDefaultFeatureToggles(
```

Comment:
TODO: Let's add an ESLint rule to forbid usage of provide (non-default) Feature Toggles inside our libs, analogically to our rules for `provideConfig` vs `provideDefaultConfig`

Reply:
done